spec/forms/idv/profile_form_spec.rb

```diff
@@ -68,13 +70,30 @@
 
     context 'when ssn is already taken by same profile' do
       it 'is valid' do
-        create(:profile, ssn: '1234', user_id: 123)
+        create(:profile, ssn: '1234', user: user)
 
         expect(subject.submit(profile_attrs.merge(ssn: '1234'))).to eq true
       end
     end
   end
 
+  describe 'dob validity' do
+    context 'when dob is not parse-able' do
+      it 'is invalid' do
+        expect(subject.submit(profile_attrs.merge(dob: '00000000'))).to eq false
+        expect(subject.errors[:dob]).to eq [t('idv.errors.bad_dob')]
+      end
+    end
+
+    context 'when dob is in the future' do
+      it 'is invalid' do
+        expect(
+          subject.submit(profile_attrs.merge(dob: (Date.today + 1).strftime('%Y-%m-%d')))
+        ).to eq false
+      end
+    end
+  end
```

Comment:
The dob tests were actually the impetus for this PR. They constitute the only new feature. I can pull all the additional validation checks into a separate PR if you like. That makes this PR a refactor only.